cli/command/container/attach.go

```diff
@@ -47,7 +62,7 @@ func runAttach(dockerCli *command.DockerCli, opts *attachOptions) error {
 	ctx := context.Background()
 	client := dockerCli.Client()
 
-	c, err := client.ContainerInspect(ctx, opts.container)
+	c, err := containerInspect(client, ctx, opts.container)
 	if err != nil {
 		return err
 	}
```

Comment:
Looks like the state check is now done twice here; once in `containerInspect()` and once below;

```go
 	if !c.State.Running {
 		return errors.New("You cannot attach to a stopped container, start it first")
 	}
 
 	if c.State.Paused {
 		return errors.New("You cannot attach to a paused container, unpause it first")
 	}
```